File: src/ReCrafted/Graphics/Display.cs
// ReCrafted © 2016-2017 Damian 'Erdroy' Korczowski

using ReCrafted.Core;

namespace ReCrafted.Graphics
{
    /// <summary>
    /// Display class.
    /// </summary>
    public static class Display
    {
        /// <summary>
        /// The client width.
        /// </summary>
        public static int ClientWidth => Game.Instance.Form.ClientSize.Width;

        /// <summary>
        /// The client height.
        /// </summary>
        public static int ClientHeight => Game.Instance.Form.ClientSize.Height;
    }
}

File: Source/Engine/ReCrafted.Engine/Core/Display.cpp
// ReCrafted (c) 2016-2020 Damian 'Erdroy' Korczowski. All rights reserved.

#include "Display.h"

uint Display::m_width;
uint Display::m_height;

void Display::ForceUpdate()
{
    uint width;
    uint height;
    Platform::GetWindowSize(Platform::GetCurrentWindow(), &width, &height);

    m_width = width;
    m_height = height;
}

float Display::GetAspectRatio()
{
    return static_cast<float>(m_width) / static_cast<float>(m_height);
}

uint Display::GetWidth()
{
    return m_width;
}

uint Display::GetHeight()
{
    return m_height;
}
